sources/packages/web/src/services/RestrictionService.ts

```diff
@@ -46,20 +46,58 @@ export class RestrictionService {
 
   public async addStudentRestriction(
     studentId: number,
-    payload: AddStudentRestrictionDTO,
+    payload: AssignRestrictionDTO,
   ): Promise<void> {
     return ApiClient.RestrictionApi.addStudentRestriction(studentId, payload);
   }
 
   public async resolveStudentRestriction(
     studentId: number,
     studentRestrictionId: number,
-    payload: UpdateRestrictionDTO,
+    payload: ResolveRestrictionDTO,
   ): Promise<void> {
     return ApiClient.RestrictionApi.resolveStudentRestriction(
       studentId,
       studentRestrictionId,
       payload,
     );
   }
+
+  public async getInstitutionRestrictions(
+    institutionId: number,
+  ): Promise<RestrictionSummaryDTO[]> {
+    return ApiClient.RestrictionApi.getInstitutionRestrictions(institutionId);
+  }
+
+  public async getInstitutionRestrictionDetail(
+    institutionId: number,
+    institutionRestrictionId: number,
+  ): Promise<RestrictionDetailDTO> {
+    return ApiClient.RestrictionApi.getInstitutionRestrictionDetail(
+      institutionId,
+      institutionRestrictionId,
+    );
+  }
+
+  public async addInstitutionRestriction(
+    institutionId: number,
+    payload: AssignRestrictionDTO,
+  ): Promise<void> {
+    return ApiClient.RestrictionApi.addInstitutionRestriction(
```

Comment:
promise id void and you are returning

Reply:
Oops! that must be await instead of return. Nice catch!